### ISSUE TITLE
[macOS 14 Beta] SDAinmatedImageRep does not play animation for GIF/APNG/WebP/HEICS (which ImageIO support)

### ISSUE THREAD

**Issue by dreampiggy** [SDWebImage/SDWebImage#3552]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Seems not only the WebP/APNG, but also GIF at all.

**Comment by dreampiggy** [comment]
Confirmed Apple's bug: FB12289074


![image](https://github.com/SDWebImage/SDWebImage/assets/6919743/29cff9b2-9907-42c8-a684-242acaab9fa8)

Demo: [NSImageViewGIFBroken.zip](https://github.com/SDWebImage/SDWebImage/files/11708028/NSImageViewGIFBroken.zip)

**Comment by dreampiggy** [comment]
Still not been fixed in 14.0 beta2 (23A5276g)
![image](https://github.com/SDWebImage/SDWebImage/assets/6919743/cb4ab0b6-e50e-4880-b33a-68e7ff46a9c2)

**Comment by dreampiggy** [comment]
Resolved by Apple in Beta5
![image](https://github.com/SDWebImage/SDWebImage/assets/6919743/a9dc3573-6860-40a4-b522-de1b8cfb8bd9)
